### PR TITLE
Use Manimpango

### PR DESCRIPTION
**Pull request by naveen521kk**
## Motivation
Stop of pangocffi, pangocairocffi and cairofffi.
Use [Manimpango](https://github.com/ManimCommunity/manimpango) instead.
The problem started with `cairocffi`, stating that it wouldn't be maintained in future https://github.com/Kozea/cairocffi/pull/173#issuecomment-749240431
Created a custom binding for Pango which seems to work, also fixes installation issues on Windows(:wink:). The only thing which doesn't work for now is the newly made `MarkupText`(sorry @PhilippImhof ). I will try to fix that soon. Need new tests for this though, removed the old ones.
Need to check the docs whether things work correctly.

## Oneline Summary of Changes
```
- Use custom bindings for Pango (:pr:`PR NUMBER HERE`)
```

## Testing Status
Worked as expected on my machine.

## Acknowledgements
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)

## Reviewer Checklist
- [ ] Newly added functions/classes are either private or have a docstring
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings
- [ ] The oneline summary has been included [in the wiki](https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release)

### CODE REVIEW

**Comment by naveen521kk** [comment]
I suspect one of the dependency has published wrong wheels or something and because of that, there is a segmentation fault of MacOS 3.8 and 3.7 need to investigate. And I don't think it is because of Github Actions update.

File: pyproject.toml [hunk 1]
```diff
@@ -35,11 +35,9 @@ pydub = "*"
...
 rich = "^6.0"
 pycairo = "^1.19"
-pangocffi = "^0.8.0"
-pangocairocffi = "^0.4.0"
-cairocffi = "^1.1.0"
-grpcio =  { version = "*", optional = true }
-grpcio-tools = { version = "*", optional = true }
+manimpango = "^0"
+grpcio =  { version = "1.33.*", optional = true }
+grpcio-tools = { version = "1.33.*", optional = true }
```
Comment (naveen521kk) [inline pyproject.toml:40]: @eulertour Is it ok to do so?

**Comment by PhilippImhof** [comment]
> The only thing which doesn't work for now is the newly made `MarkupText`(sorry @PhilippImhof ). I will try to fix that soon. 

AFAICS `MarkupText` does not use other methods from Pango / Cairo than `Text`, so I would say there is no reason why `MarkupText` could not work with it. (I see that you changed the names of the old methods, but that should be easy to fix.)

File: pyproject.toml (hunk 1 above)
Comment (eulertour) [reply pyproject.toml:40]: Yes that's fine

**Comment by PhilippImhof** [comment]
I can confirm that this works fine on Mac OS Catalina with Python 3.9 and Pango 1.48 installed

**Comment by naveen521kk** [comment]
> I can confirm that this works fine on Mac OS Catalina with Python 3.9 and Pango 1.48 installed

I have just released it. It will take some time for wheels to build and uploaded to PyPi. After that, `poetry.lock` needs to be updated.

**Comment by naveen521kk** [comment]
This is ready for Review.

**Comment by behackl** [comment]
I've played around with this a little, some observations (I'm on Ubuntu):

- When working directly from the interpreter, I get
```py
>>> from manim import *
>>> Text("asdf")
[12/29/20 13:27:54] INFO     Text now uses Pango for rendering. In case of problems, the old implementation is available as CairoText.                                                                                      text_mobject.py:717
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 72: non-double matrix element
Fontconfig error: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 72: non-double matrix element
Fontconfig warning: "/etc/fonts/conf.d/10-scale-bitmap-fonts.conf", line 80: saw unknown, expected number
Text('asdf')
```
  and the same thing for `MarkupText`. This also creates a directory `fontconfig` in my current working directory with many files in it.

- Trying to use `Code` leads to an error:
```py
(manim-jRHFv4qq-py3.7) behackl@MR-SKELTAL:~/code/manim/playground$ manim av.py TextTest -qm
[12/29/20 13:34:01] INFO     Text now uses Pango for rendering. In case of problems, the old implementation is available as CairoText.                                             text_mobject.py:717



Traceback (most recent call last):
  File "/home/behackl/code/manim/manim/__main__.py", line 95, in main
    scene.render()
  File "/home/behackl/code/manim/manim/scene/scene.py", line 165, in render
    self.construct()
  File "av.py", line 5, in construct
    c = Code("./ag.py")
  File "/home/behackl/code/manim/manim/mobject/svg/code_mobject.py", line 188, in __init__
    self.code = self.gen_colored_lines()
  File "/home/behackl/code/manim/manim/mobject/svg/code_mobject.py", line 316, in gen_colored_lines
    stroke_width=self.stroke_width,
  File "/home/behackl/code/manim/manim/mobject/svg/text_mobject.py", line 441, in __init__
    self.lines_text = Text(lines_str, **config)
  File "/home/behackl/code/manim/manim/mobject/svg/text_mobject.py", line 759, in __init__
    file_name = self.text2svg()
  File "/home/behackl/code/manim/manim/mobject/svg/text_mobject.py", line 964, in text2svg
    self.text,
  File "cmanimpango.pyx", line 234, in manimpango.cmanimpango.text2svg
  File "/usr/lib/python3.7/xml/sax/saxutils.py", line 27, in escape
    data = data.replace("&", "&amp;")
TypeError: a bytes-like object is required, not 'str'
```
This is from trying to run
```py
from manim import *

class TextTest(Scene):
    def construct(self):
        c = Code("./ag.py")
        self.play(ShowCreation(c))
```

which simply uses

```py
from manim import *

class SimpleScene(Scene):
    def construct(self):
        s = Square()
        self.play(ShowCreation(s))
```
as `ag.py`.

- Simply using `Text` seems to work just fine, great job!!

**Comment by naveen521kk** [comment]
Oh! I didn't notice that. I see where the bug is. I will quickly patch it, in an hour or so. Meanwhile, this time after update, there would be a different wheel released for linux which latest Pango version. I will also update this PR so that it uses that version as the minimum one.

**Comment by naveen521kk** [comment]
Also, I will update the docs about loading from a TTF file, also specify it is only available for linux.

**Comment by naveen521kk** [comment]
Also, I think @behackl  would need to update the docker image so that this wheel works. I have tested this on default python file but it fails on `slim` version of the same docker image. Either we should try fixing it or use python default as our base image.

**Comment by behackl** [comment]
> Also, I think @behackl would need to update the docker image so that this wheel for ManimPango works. I have tested this on default python file but it fails on `slim` version of the same docker image. Either we should try fixing it or use python default as our base image.

Good to know, thank you. I'll investigate docker a bit later.

**Comment by naveen521kk** [comment]
Also, when can it be merged? Before or after this month's release?

**Comment by naveen521kk** [comment]
This PR is ready. Just some docs on Linux only feature of font from file is remaining.
I would like to merge this before the release. Because without this it will be another solving the horrible DLL HELL for another month 😨

**Comment by kolibril13** [comment]
I just tested this branch on Ubuntu 20.04.
Text works, and also when I run it from the python console 
```py
>>> from manim import *
>>> Text("Hello")
[12/31/20 10:09:17] INFO     Text now uses Pango for         text_mobject.py:716
                             rendering. In case of problems,                    
                             the old implementation is                          
                             available as CairoText.                            
```
I do not get any errors.
Further, also the `Code` example from @behackl works now on my machine:

https://user-images.githubusercontent.com/44469195/103403747-83c39800-4b51-11eb-9ea6-f366baf167d4.mp4

**Review by kolibril13 — APPROVED** [review]

**Comment by behackl** [comment]
A bit late, but I still managed: I've looked into potential issues with docker, found that our `Dockerfile` does not require the workaround for generating the ffi bindings any longer (removed via ad9fa7e4), and then tested the Code example from above. I *did* need to pass `font="Monospace"` to `Code`, but the fact that docker manages to render this as

https://user-images.githubusercontent.com/11851593/103406019-77433d80-4b59-11eb-9330-3258c62095b9.mp4

is very good; no further adaptions seem to be required.

I'm positively leaning towards merging this before the release, I'll test it some more before approving.

**Comment by behackl** [comment]
I've updated the troubleshooting page via cdb91780.

**Review by behackl — APPROVED** [review]
Given the fact that this 

- (has the potential to) significantly simplify the installation procedure and free us from pango-related issues (especially for Windows users),
- replaces an interface that will not be maintained any longer soon,
- and works really well,

I will go ahead and approve this, with the intention of merging it before the release. I do believe that this time around, the switch will be smoother than when switching from `CairoText` to `PangoText` -- but in case something seriously breaks, nothing stops us from releasing a bugfix version earlier than our usual release schedule.

**Comment by naveen521kk** [comment]
Let both CairoText and ManimText be there as it is and can be seen in next release. 
I'm merging this.

**Pull request merged**